### PR TITLE
Address clock drift.

### PR DESCRIPTION
**Pull request by jiceatscion**
This is almost identical to what I [added to the DP draft](https://github.com/scionassociation/scion-dp_I-D/pull/19). I also fixed a weird sentence about the routers dropping packets with expired hops.

The reviewers attention is kindly directed to section 2.2.3.

Addresses #18

### CODE REVIEW

**Review by nicorusti — CHANGES_REQUESTED** [review]
Hi @jiceatscion there are a few things worth addressing, the biggest one regards the new section on clock drift. Let's perhaps discuss this directly after you take a look at my comments.

File: draft-dekater-scion-controlplane.md [hunk 1]
```diff
@@ -991,8 +1001,8 @@ The following code block defines the hop field component `HopField` in Protobuf
...
-- `exp_time`: The 8-bit encoded expiration time of the hop field, indicating how long the hop field is valid. This value is an offset relative to the PCB creation timestamp set in the PCB's segment information component (see also [](#seginfo)). By combining these two values, the AS can compute the absolute expiration time of the hop field. Data-plane packets that use the hop field after the expiration time MUST be dropped.
-- `mac`: The message authentication code (MAC) used in the data plane to verify the hop field. The SCION Data Plane Specification provides a detailed description of the computation of the MAC and the verification of the hop field in the data plane.
+- `exp_time`: The 8-bit encoded expiration time of the hop field, indicating its validity. This field expresses a duration in seconds according to the formula: `duration = (1 + exp_time) * (24*60*60/256)`. The minimum duration is therefore 337.5 s. This duration is relative to the PCB creation timestamp set in the PCB's segment information component (see also [](#seginfo)). By combining these two values, an AS can compute the absolute expiration time of the hop field.
```
Comment (matzf) [inline draft-dekater-scion-controlplane.md:1004]: > By combining these two values, an AS can compute the absolute expiration time of the hop field.

Nit: _anyone_ can compute the expiration time, not only "an AS". E.g. endpoints might have to do this to check if a path is still usable.

File: draft-dekater-scion-controlplane.md [hunk 2]
```diff
@@ -1173,16 +1196,16 @@ PCBs are propagated in batches to each connected downstream AS at a fixed freque
...
+The PCB contains the version numbers of the trust root configuration(s) (TRC) and certificate(s) that must be used to verify its signatures. This enables the control service to check whether it has the relevant TRC(s) and certificate(s); if not, they can be requested from the control service of the sending AS.
+2. As core beaconing is based on flooding PCBs, it is necessary to avoid loops during path creation. The control service of core ASes MUST therefore check whether the PCB includes duplicate hop entries created by the core AS itself or by other ASes. If so, the PCB MUST be discarded in order to avoid loops. Additionally, core ASes could forbid, that is, not propagate, beacons containing path segments that traverse the same ISD more than once. **Note:** Where loops must always be avoided, it is a policy decision to forbid ISD double-crossing. It can be legitimate to cross the same ISD multiple times: For example, if the ISD spans a large geographical area, a path transiting another ISD may constitute a shortcut. However, it is up to each core AS to decide whether it wants to allow this.
```
[the remark this answers is not in the document]
Comment (jiceatscion) [reply draft-dekater-scion-controlplane.md:1208]: as we discussed. I added a PCB validity section and moved, and simplify, the clock accuracy section.

File: draft-dekater-scion-controlplane.md (hunk 1 above)
Comment (jiceatscion) [reply draft-dekater-scion-controlplane.md]: done.

File: draft-dekater-scion-controlplane.md [hunk 3]
```diff
@@ -1069,6 +1079,19 @@ On code-level and in Protobuf message format, extensions are specified as follow
...
+
+The unit of a segment's time-to-live is 5 minutes and 37 seconds (and 500 ms), or the equivalent of 5 hops. Given the above constraints, it is unreasonable to create a segment with a short time-to-live, while each additional time-to-live unit adds more than 5 minutes. As a result, a clock drift of up to 1 minute can be safely neglected.
+
+The control service and its clients authenticate each-other according to their respective AS's certificate. Path segments are authenticated based on the certificates of the ASes that they refer to. The time validity of a certificate is affected by the offset between verifier and originator clocks. The expiration of a SCION AS certificate typically ranges from 3h to 5 years. As a result, a time offset measured in minutes is immaterial.
+
+Each administrator of a SCION router or core control service is responsible for maintaining sufficient clock accuracy. No particular method is assumed by this specification.
```
[the remark this answers is not in the document]
Comment (jiceatscion) [reply draft-dekater-scion-controlplane.md]: I rewrote/moved based on these comments and our subsequent discussions. I leaned in the direction of less: the reader needs to be aware of the forces at play but can draw their own conclusions. They are fairly obvious but highly dependent on use case and policies. If we try to give advice we end-up with a complete administrator's manual. Not what we want here.

Tentatively resolving now.

[the remark this answers is not in the document]
Comment (nicorusti) [reply draft-dekater-scion-controlplane.md]: Thanks a lot, looks good now!

**Review by nicorusti — APPROVED** [review]
Thanks a lot for restructuring the paragraph, looks good!

**Pull request merged**